src/http/client/client.cr

```diff
@@ -303,7 +304,8 @@ class HTTP::Client
     request.headers["User-agent"] ||= "Crystal"
     request.to_io(socket)
     socket.flush
-    HTTP::Response.from_io(socket) do |response|
+    ignore_body = request.method == "HEAD"
```

Comment:
This check is duplicated, please extract it to a method somewhere alike `mandatory_body?`, so that when we discover new cases there's a single place to update.